catalyst/loggers/comet.py

```diff
@@ -0,0 +1,264 @@
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs image to the logger."""
+        self.image_name = tag
+        key_parameters = [stage_key, loader_key, scope]
+        # Removes all None values from list.
+        passed_key_parmeters = [key_parameter for key_parameter in key_parameters if key_parameter]
+        if len(passed_key_parmeters) == 1:
+            keys_prefix = passed_key_parmeters[0]
+            self.image_name = f"{keys_prefix}_{tag}"
+        elif len(passed_key_parmeters) > 1:
+            keys_prefix = '_'.join(passed_key_parmeters)
+            self.image_name = f"{keys_prefix}_{tag}"
+        self.experiment.log_image(image, name=self.image_name, step=global_batch_step)
+
+    def log_hparams(
+        self,
+        hparams: Dict,
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        stage_key: str = None,
+    ) -> None:
+        """Logs hyperparameters to the logger."""
+        key_parameters = [stage_key, scope]
+        passed_key_parmeters = [key_parameter for key_parameter in key_parameters if key_parameter]
+        if len(passed_key_parmeters) == 1:
+            keys_prefix = passed_key_parmeters[0]
+            self.experiment.log_parameters(hparams, prefix=keys_prefix)
+        elif len(passed_key_parmeters) > 1:
+            keys_prefix = '_'.join(passed_key_parmeters)
+            self.experiment.log_parameters(hparams, prefix=keys_prefix)
+        else:
+            self.experiment.log_parameters(hparams)
+
+    def log_artifact(
+        self,
+        tag: str = "artifact",
+        artifact: object = None,
+        path_to_artifact: str = None,
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs artifact (arbitrary file like audio, video, model weights) to the logger."""
+        if path_to_artifact or artifact:
```

Comment:
Since we're passing `passed_key_parameters` in both cases, we don't need this `if` statement here. 

To be explicit about how these parameters are being used I would suggest renaming `key_parameters` to something like `metadata_parameters` and `passed_key_parameters` to `passed_metadata_parameters`